src/include/rbd_types.h

```diff
@@ -106,4 +107,9 @@ struct rbd_obj_header_ondisk {
 	struct rbd_obj_snap_ondisk snaps[0];
 } __attribute__((packed));
 
+enum group_image_link_state {
+	LINK_NORMAL,
```

Comment:
Minor: enum values should include enum name: a la GROUP_IMAGE_LINK_STATE_XYZ. "normal" and "dirty" are not too descriptive -- seems like DIRTY implies ATTACHING and NORMAL implies ATTACHED.